### PR TITLE
allow jsx identifiers to start with keywords

### LINKED ISSUE #4644: Jsx attributes that start with keyword are not handled correctly
Attempt to compile code below issues a number of errors:
```ts
declare var React: any;
<foo class-id/>
```

```
node built\local\tsc.js  --jsx preserve ..\foo.tsx
../foo.tsx(2,11): error TS1003: Identifier expected.
../foo.tsx(2,15): error TS1109: Expression expected.
../foo.tsx(3,1): error TS1109: Expression expected.
```

Online JSX compiler handles it correctly

### PR DESCRIPTION
**Pull request by vladima**
fixes #4644

### CODE REVIEW

File: src/compiler/parser.ts [hunk 1]
```diff
@@ -12,6 +12,10 @@ namespace ts {
     export function createNode(kind: SyntaxKind): Node {
         return new (getNodeConstructor(kind))();
     }
+    
+    export function tokenIsIdentifierOrKeyword(token: SyntaxKind): boolean {
```
Comment (DanielRosenwasser) [inline src/compiler/parser.ts]: This should be exported from the scanner (and marked internal).

File: src/compiler/parser.ts [hunk 2]
```diff
@@ -4102,7 +4106,7 @@ namespace ts {
         }
 
         function isIdentifierOrKeyword() {
-            return token >= SyntaxKind.Identifier;
+            return tokenIsIdentifierOrKeyword(token);
```
Comment (mhegazy) [inline src/compiler/parser.ts]: do you need this function then?

**Comment by DanielRosenwasser** [comment]
Can you add an example of the property being a keyword entirely? Like `class="FooAligned"`.

**Comment by mhegazy** [comment]
:+1:  i would merge this into master and not release-1.6

**Pull request merged**